src/main/menu/electron-menu-items.injectable.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import { Injectable, lifecycleEnum } from "@ogre-tools/injectable";
+import { computed, IComputedValue } from "mobx";
+import type { LensMainExtension } from "../../extensions/lens-main-extension";
+import extensionsInjectable from "../../extensions/extensions.injectable";
+import type { MenuRegistration } from "./menu-registration";
+
+const electronMenuItemsInjectable: Injectable<
+  IComputedValue<MenuRegistration[]>,
+  { extensions: IComputedValue<LensMainExtension[]> }
+> = {
+  lifecycle: lifecycleEnum.singleton,
+
+  getDependencies: di => ({
+    extensions: di.inject(extensionsInjectable),
+  }),
+
+  instantiate: ({ extensions }) =>
+    computed(() => extensions.get().flatMap(extension => extension.appMenus)),
```

Comment:
what about disabled extensions?

Reply:
Just noticed that there is something weird happening here. Disabled extensions work already, but it seems that is not enough. We need to be sure that `.init()` has been called for extension as well. 

I have required changes done already and pushing those here tomorrow.

Reply:
Hmm. In general, with DI we will have more guarantees that temporal requirements, such as `.init()` this or `.createInstance()` that, will have been called when expected. Good.